build.gradle

```diff
@@ -0,0 +1,35 @@
+plugins {
+	id 'java'
+	id 'org.springframework.boot' version '3.1.5'
+	id 'io.spring.dependency-management' version '1.1.3'
+}
+
+group = 'daronek6'
+version = '0.0.1-SNAPSHOT'
+
+java {
+	sourceCompatibility = '17'
+}
+
+configurations {
+	compileOnly {
+		extendsFrom annotationProcessor
+	}
+}
+
+repositories {
+	mavenCentral()
+}
+
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	compileOnly 'org.projectlombok:lombok'
+	runtimeOnly 'com.h2database:h2'
```

Comment:
Fajnie, że używasz bazy in-memory :+1: W jaki sposób ustawiłbyś by aplikacja po wdrożeniu nie korzystała z H2 a np. z Postgresa? Co musiałbyś zrobić? Dalibyśmy rady mieć np. 4 środowiska i na nich być połączonym do różnych baz? Jak moglibyśmy to skonfigurować?

Reply:
Nigdy tego nie robiłem, ale z tego co się dowiedziałem, to zmiana z H2 na np. Postgresa wymagałoby zmiany `runtimeOnly 'com.h2database:h2'` na `implementation 'org.postgresql:postgresql'` i dodania konfiguracji bazy w pliku application.properties. Plik konfiguracyjny zawiera między innymi sterownik bazy, adres dostępu do bazy, logi i hasło użytkownika, port i inne możliwe ustawienia.

Jeśli chodzi o środowiska, to należałoby stworzyć oddzialne profile w postaci plików konfiguracyjnych w postaci application-{nazwaProfilu}.properties, w których znajdowałyby się konfiguracje do innych baz. Zakładam, że różne bazy nie oznacza różne rodzaje baz np MySQL i PostgeSQL, a poprostu rózne bazy tego samego rodzaju np. baza produkcji i testów. Następnie w pliku konfiguracyjnym application.properties ustalić z którego profilu korzystamy `spring.profiles.active={nazwaProfilu}`. Można jeszcze wykorzystać adnotacje w kodzie, i wskazać który profil co powinien robić https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.profiles.

Reply:
Fajnie, że doczytałeś :+1: